### PR TITLE
[menu] Reimplement the bottom sheet using native languages

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Closes ENG-7280.
Reimplemented the bottom sheet using native languages and removed reanimated from vendored modules. 
By doing that we improve the build speed.

# How

- On Android, I've used the material design bottom sheet. We may want to copy the implementation from the original repo to ours to avoid extra dependency.
- On iOS, I've copied the implementation from https://github.com/applidium/OverlayContainer. I managed to test a couple of different libraries and that one works the best with RN. 

# Test Plan

- bare-expo on both platforms ✅

### CODE REVIEW

File: packages/expo-dev-menu/ios/OverlayContainer/Internal/Extensions/UIScrollViiew+Utils.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+}
+
+
+extension UIScrollView {
+    
+    var oc_adjustedContentInset: UIEdgeInsets {
+        if #available(iOS 11.0, *) {
+            return self.adjustedContentInset
+        } else {
```
Comment (douglowder) [inline packages/expo-dev-menu/ios/OverlayContainer/Internal/Extensions/UIScrollViiew+Utils.swift:35]: I don't believe we support anything earlier than 12.4, or maybe 13.0 now, so not sure this is needed.

File: packages/expo-dev-menu/ios/OverlayContainer/Internal/Extensions/UIScrollViiew+Utils.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            // Fallback on earlier versions
+            return self.contentInset
+        }
```
Comment (douglowder) [inline packages/expo-dev-menu/ios/OverlayContainer/Internal/Extensions/UIScrollViiew+Utils.swift:38]:
```suggestion
        return self.adjustedContentInset
```

File: packages/expo-dev-menu/ios/OverlayContainer/Internal/HeightContraintOverlayTranslationController.swift [hunk 3]
```diff
@@ -0,0 +1,296 @@
+//
+//  HeightConstraintOverlayTranslationController.swift
+//  OverlayContainer
+//
+//  Created by Gaétan Zanella on 29/11/2018.
```
Comment (douglowder) [inline packages/expo-dev-menu/ios/OverlayContainer/Internal/HeightContraintOverlayTranslationController.swift:5]: If we do not own the copyright, we should reproduce the license (MIT, etc.) here

File: packages/expo-dev-menu/ios/OverlayContainer/Internal/InterruptibleAnimatorOverlayContainerTransitionCoordinator.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
+//
+//  InterruptibleAnimatorOverlayContainerTransitionCoordinator.swift
+//  OverlayContainer
+//
+//  Created by Gaétan Zanella on 28/11/2018.
```
Comment (douglowder) [inline packages/expo-dev-menu/ios/OverlayContainer/Internal/InterruptibleAnimatorOverlayContainerTransitionCoordinator.swift:5]: Same comment on license as above

**Review by douglowder — CHANGES_REQUESTED** [review]
Just a few minor questions and one suggestion, and fix Swiftlint issues, looks good otherwise

File: packages/expo-dev-menu/ios/DevMenuWindow.swift [hunk 5]
```diff
@@ -2,25 +2,111 @@
 
 import UIKit
 
-class DevMenuWindow: UIWindow {
-  private let manager: DevMenuManager
+extension UIView {
+  func firstSubView<T: UIView>(ofType type: T.Type) -> T? {
```
Comment (tsapeta) [inline packages/expo-dev-menu/ios/DevMenuWindow.swift:6]:
```suggestion
  func firstSubview<T: UIView>(ofType type: T.Type) -> T? {
```
Also I'd feel more comfortable with not using the extension when there is another solution like a top-level private function 😅

File: packages/expo-dev-menu/ios/OverlayContainer/Internal/Extensions/UIScrollViiew+Utils.swift (hunk 1 above)
Comment (lukmccall) [reply packages/expo-dev-menu/ios/OverlayContainer/Internal/Extensions/UIScrollViiew+Utils.swift:35]: It's not our code and I don't want to modify that.

**Review by tsapeta — APPROVED** [review]

**Pull request merged**